front/lib/api/assistant/conversation.ts

```diff
@@ -1317,6 +1318,37 @@ export async function retryAgentMessage(
         t,
       });
 
+      // Find all sibling messages (messages with the same parentId)
+      const siblingMessages = await Message.findAll({
+        where: {
+          conversationId: conversation.id,
+          workspaceId: auth.getNonNullableWorkspace().id,
+          parentId: messageRow.parentId,
+          visibility: "visible",
+        },
+        transaction: t,
+      });
+
+      // Find the maximum rank among all sibling messages
+      const maxSiblingRank = Math.max(...siblingMessages.map((m) => m.rank));
+
+      // Hide all messages with ranks greater than the maximum sibling rank
+      await Message.update(
+        { visibility: "hidden" },
+        {
+          where: {
+            conversationId: conversation.id,
+            workspaceId: auth.getNonNullableWorkspace().id,
+            rank: {
+              [Op.gt]: maxSiblingRank,
+            },
+            visibility: "visible", // Only hide visible messages (not already deleted/hidden)
+          },
+          transaction: t,
+          validate: false, // Skip validation hooks since we're only updating visibility
```

Comment:
Should we remove that? You added it cause it's going to be faster I assume, but do we want to use this pattern?